### PR TITLE
Added extension exposing the spread WKWebView

### PR DESCRIPTION
**Pull request by cbaltzer**
Allows client integrations to access the internal WKWebView in order to add custom user scripts, etc. Added as an extension rather than modifying the existing structure.

### CODE REVIEW

**Comment by mickael-menu** [comment]
I'm thinking that maybe we don't need to expose the full web view, but maybe only the `webView.configuration.userController`, to be able to inject scripts. This would prevent apps from messing up too much with the web views and its delegates at this point. Do you think this would be enough for your needs?

Also I'm afraid accessing the current web view would put a lot of burden on reading apps to synchronize their scripts, because the navigator creates new web views all the time and several web views are running at the same time. Instead we could have a new delegate API [`EPUBNavigatorDelegate::setupWebViewScripts(userContentController: WKUserContentController)`](https://github.com/readium/r2-navigator-swift/blob/0f1e8839c23b6ab91ccd39b26ada519c368e0149/r2-navigator-swift/EPUB/EPUBNavigatorViewController.swift#L19) called [from here](https://github.com/readium/r2-navigator-swift/blob/0f1e8839c23b6ab91ccd39b26ada519c368e0149/r2-navigator-swift/EPUB/EPUBNavigatorViewController.swift#L666).

And maybe another one `EPUBNavigatorDelegate::resource(link: Link, didReceiveScriptMessage: WKScriptMessage)` could be useful for callbacks from custom scripts, if the `message.name` is unknown here: https://github.com/readium/r2-navigator-swift/blob/0f1e8839c23b6ab91ccd39b26ada519c368e0149/r2-navigator-swift/EPUB/EPUBSpreadView.swift#L416-L418

**Comment by cbaltzer** [comment]
> I'm thinking that maybe we don't need to expose the full web view, but maybe only the `webView.configuration.userController`, to be able to inject scripts. This would prevent apps from messing up too much with the web views and its delegates at this point. Do you think this would be enough for your needs?

Not totally sure just yet, but I suspect that would be okay. 

> Also I'm afraid accessing the current web view would put a lot of burden on reading apps to synchronize their scripts, because the navigator creates new web views all the time and several web views are running at the same time. Instead we could have a new delegate API [`EPUBNavigatorDelegate::setupWebViewScripts(userContentController: WKUserContentController)`](https://github.com/readium/r2-navigator-swift/blob/0f1e8839c23b6ab91ccd39b26ada519c368e0149/r2-navigator-swift/EPUB/EPUBNavigatorViewController.swift#L19) called [from here](https://github.com/readium/r2-navigator-swift/blob/0f1e8839c23b6ab91ccd39b26ada519c368e0149/r2-navigator-swift/EPUB/EPUBNavigatorViewController.swift#L666).

I ran into this issue today, this is a great idea. We need scripts injected into every view, so having a delegate call during their creation would allow that. 

> And maybe another one `EPUBNavigatorDelegate::resource(link: Link, didReceiveScriptMessage: WKScriptMessage)` could be useful for callbacks from custom scripts, if the `message.name` is unknown here:
> 
> https://github.com/readium/r2-navigator-swift/blob/0f1e8839c23b6ab91ccd39b26ada519c368e0149/r2-navigator-swift/EPUB/EPUBSpreadView.swift#L416-L418

Also a great suggestion.

**Comment by cbaltzer** [comment]
@mickael-menu Check the latest commit for the delegate additions. I wasn't sure what the proper `link` should be for the resource delegate so let me know about that.

File: r2-navigator-swift/EPUB/EPUBNavigatorViewController.swift [hunk 1]
```diff
@@ -635,6 +642,10 @@ extension EPUBNavigatorViewController: EPUBSpreadViewDelegate {
         present(viewController, animated: true)
     }
 
+    func spreadView(_ spreadView: EPUBSpreadView, didReceiveMessage message: WKScriptMessage) {
+        delegate?.resource(link: spreadView.spread.leading, didReceiveMessage: message)
```
Comment (cbaltzer) [inline r2-navigator-swift/EPUB/EPUBNavigatorViewController.swift:646]: Not sure if this is the correct link.

Comment (mickael-menu) [reply r2-navigator-swift/EPUB/EPUBNavigatorViewController.swift]: You're right, it's difficult to get the proper link at this point so I mirrored the API of WebKit instead. You can always get the current resource from the JavaScript.

**Review by mickael-menu — APPROVED** [review]
@cbaltzer I made some minor fixes in the delegate APIs name and removed the exposition of the current web view. You can test it out from the `develop` branch, let me know if this is enough to answer your use cases.

**Comment by mickael-menu** [comment]
@cbaltzer This is part of the latest 2.0.0-beta.2 release

**Pull request merged**